compiler/src/dotty/tools/dotc/transform/ElimErasedValueType.scala

```diff
@@ -87,7 +87,10 @@ class ElimErasedValueType extends MiniPhase with InfoTransformer { thisPhase =>
       val site = root.thisType
       val info1 = site.memberInfo(sym1)
       val info2 = site.memberInfo(sym2)
-      if (!info1.matchesLoosely(info2))
+      if (!info1.matchesLoosely(info2) &&
+          !(sym1.name == nme.apply &&
```

Comment:
I did not understand why this additional condition was needed.To make it clearer, define it as an auxiliary def and add a doc comment?

Reply:
Done.